build.gradle

```diff
@@ -498,24 +543,26 @@ project(':iceberg-hive-metastore') {
       exclude group: 'org.slf4j', module: 'slf4j-log4j12'
     }
 
-    testCompile project(path: ':iceberg-api', configuration: 'testArtifacts')
+    testImplementation project(path: ':iceberg-api', configuration: 'testArtifacts')
   }
 }
 
 project(':iceberg-mr') {
   configurations {
-    testCompile {
+    testImplementation {
       exclude group: 'org.apache.parquet', module: 'parquet-hadoop-bundle'
     }
   }
 
   dependencies {
-    compile project(':iceberg-api')
-    compile project(':iceberg-core')
-    compile project(':iceberg-data')
-    compile project(':iceberg-hive-metastore')
-    compile project(':iceberg-orc')
-    compile project(':iceberg-parquet')
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    api project(':iceberg-api')
+    implementation project(':iceberg-common')
+    implementation project(':iceberg-core')
+    implementation project(':iceberg-data')
```

Comment:
This should also be an API dependency because the `InputFormat` produces Iceberg classes in iceberg-data.

Reply:
good point, updated this and `iceberg-spark`/`iceberg-flink` to an `api` dependency, since those were also using iceberg-data class in their API